practicas/dianasalgado22/P1/01_niveles_linguisticos.py

```diff
@@ -0,0 +1,689 @@
+    if status_code != http.HTTPStatus.OK:
+        print(f"ERROR on {iso_lang} :(")
+        return ""
+    return response.text
+
+
+# -
+
+def parse_response(response: str) -> dict:
+    """Parse text response from ipa-dict to python dict
+
+    Each row have the format:
+    [WORD][TAB]/[IPA]/(, /[IPA]/)?
+
+    Parameters
+    ----------
+    response: str
+        ipa-dict raw text
+
+    Returns
+    -------
+    dict:
+        A dictionary with the word as key and the phonetic
+        representations as value
+    """
+    ipa_list = response.rstrip().split("\n")
+    result = {}
+    for item in ipa_list:
+        if item == '':
+            continue
+        item_list = item.split("\t")
+        result[item_list[0]] = item_list[1]
+    return result
+
+
+def get_ipa_transcriptions(word: str, dataset: dict) -> list[str]:
+    """Search for a word in an IPA phonetics dict
+
+    Given a word this function return the IPA transcriptions
+
+    Parameters:
+    -----------
+    word: str
+        A word to search in the dataset
+    dataset: dict
+        A dataset for a given language code
+
+    Returns
+    -------
+    list[str]:
+        List with posible transcriptions if any,
+        else an empty list
+    """
+    return dataset.get(word.lower(), "").split(", ")
+
+
+# #### Obteniendo todos los datos
+
+IPA_URL = "https://raw.githubusercontent.com/open-dict-data/ipa-dict/master/data/{lang}.txt"
+
+lang_codes = {
+    "ar": "Arabic (Modern Standard)",
+    "de": "German",
+    "en_UK": "English (Received Pronunciation)",
+    "en_US": "English (General American)",
+    "eo": "Esperanto",
+    "es_ES": "Spanish (Spain)",
+    "es_MX": "Spanish (Mexico)",
+    "fa": "Persian",
+    "fi": "Finnish",
+    "fr_FR": "French (France)",
+    "fr_QC": "French (Québec)",
+    "is": "Icelandic",
+    "ja": "Japanese",
+    "jam": "Jamaican Creole",
+    "km": "Khmer",
+    "ko": "Korean",
+    "ma": "Malay (Malaysian and Indonesian)",
+    "nb": "Norwegian Bokmål",
+    "nl": "Dutch",
+    "or": "Odia",
+    "ro": "Romanian",
+    "sv": "Swedish",
+    "sw": "Swahili",
+    "tts": "Isan",
+    "vi_C": "Vietnamese (Central)",
+    "vi_N": "Vietnamese (Northern)",
+    "vi_S": "Vietnamese (Southern)",
+    "yue": "Cantonese",
+    "zh_hans": "Mandarin (Simplified)",
+    "zh_hant": "Mandarin (Traditional)"
+}
+iso_lang_codes = list(lang_codes.keys())
+
+
+def get_corpora() -> dict:
+    """Download corpora from ipa-dict github
+
+    Given a list of iso lang codes download available datasets.
+
+    Returns
+    -------
+    dict
+        Lang codes as keys and dictionary with words-transcriptions
+        as values
+    """
+    return {
+        code: parse_response(download_ipa_corpus(code))
+         for code in iso_lang_codes
+        }
+
+
+data = get_corpora()
+
+# #### Sistema de búsqueda  
+
+# +
+from Levenshtein import distance as levenshtein_distance
+
+def buscar_palabras_similares(palabra, dicc , top_n=3):
```

Comment:
Igual recomiendo agregar el typing 

```suggestion
def buscar_palabras_similares(palabra: str, dicc: dict , top_n: int = 3) -> list[str]:
```